### PR TITLE
adding docs about virtual forms to the forms guide

### PR DESCRIPTION
**Pull request by jwoertink**
This includes docs about Virtual Forms.

### CODE REVIEW

File: source/guides/saving-with-forms.html.md.erb [hunk 1]
```diff
@@ -549,6 +549,45 @@ class SignUps::NewPage < MainLayout
...
+      text_input f.query
+
+      label_for f.active
+      checkbox f.active
+      submit "Filter Results"
+    end
+  end
+end
+```
+
```
Comment (paulcsmith) [inline source/guides/saving-with-forms.html.md.erb:598]: Could you add a simpler example of using the form? Like adding a `submit` method and maybe some validations or something?

```crystal
def submit
  validate_required query
  yield form, UserQuery.new.name.ilike(query.value).beginned_at.gt(begin_date.value).active(active.value)
end
```

Comment (paulcsmith) [reply source/guides/saving-with-forms.html.md.erb:598]: And maybe mention that this a convention to return the form and a result, but you can name the method and return whatever you want

**Review by paulcsmith — COMMENTED** [review]
So stoked to have this in. A lot of people ask about this so it'll be helpful to have something to point them to 👍

File: source/guides/saving-with-forms.html.md.erb (hunk 1 above)
Comment (jwoertink) [reply source/guides/saving-with-forms.html.md.erb:598]: Alright, I updated that. I don't know that the example is "easier", but I think it's clearer. I added the full circle to include going back to the action. That was the part that tripped me up using it initially. Let me know if you see anything else.

**Review by paulcsmith — APPROVED** [review]
Looks good :D

**Pull request merged**